packages/watt-pprof-capture/index.js

```diff
@@ -4,65 +4,158 @@ const pprof = require('@datadog/pprof')
 const {
   ProfilingAlreadyStartedError,
   ProfilingNotStartedError,
+  ProfilingJobAlreadyStartedError,
+  ProfilingJobNotStartedError,
+  ProfilingJobAlreadyPausedError,
+  ProfilingJobMissingOptionError,
   NoProfileAvailableError,
 } = require('./lib/errors')
 
 const kITC = Symbol.for('plt.runtime.itc')
 
-let isCapturing = false
 let latestProfile = null
-let captureInterval = null
+
+let isProfiling = false
+let isProfilingJobStarted = false
+let isProfilingJobPaused = false
+let profilingJobOptions = null
+let profilingJobInterval = null
 
 // Keep trying until ITC is available. This is needed because preloads run
 // before the app thread initialization, so globalThis.platformatic.messaging
 // and ITC don't exist yet.
 const registerInterval = setInterval(() => {
   if (globalThis[kITC]) {
     globalThis[kITC].handle('getLastProfile', getLastProfile)
+    globalThis[kITC].handle('startProfilingJob', startProfilingJob)
+    globalThis[kITC].handle('stopProfilingJob', stopProfilingJob)
     globalThis[kITC].handle('startProfiling', startProfiling)
     globalThis[kITC].handle('stopProfiling', stopProfiling)
     clearInterval(registerInterval)
   }
 }, 10)
 
-function rotateProfile () {
-  // `true` immediately restarts profiling after stopping
-  latestProfile = pprof.time.stop(true)
-}
-
 function startProfiling (options = {}) {
-  if (isCapturing) {
+  if (isProfiling) {
     throw new ProfilingAlreadyStartedError()
   }
-  isCapturing = true
+  isProfiling = true
 
-  pprof.time.start(options)
+  if (isProfilingJobStarted) {
+    pauseProfilingJob()
+  }
 
-  // Set up profile window rotation if durationMillis is provided
-  const timeout = options.durationMillis
-  if (timeout) {
-    captureInterval = setInterval(rotateProfile, timeout)
-    captureInterval.unref()
+  try {
+    pprof.time.start(options)
+  } catch (err) {
+    if (isProfilingJobPaused) {
+      resumeProfilingJob()
+    }
+    isProfiling = false
+
+    throw err
   }
 }
 
 function stopProfiling () {
-  if (!isCapturing) {
+  if (!isProfiling) {
     throw new ProfilingNotStartedError()
   }
-  isCapturing = false
 
-  clearInterval(captureInterval)
-  captureInterval = null
+  let profile = null
 
-  latestProfile = pprof.time.stop()
-  return latestProfile.encode()
+  try {
+    profile = pprof.time.stop().encode()
+  } finally {
+    isProfiling = false
+
+    if (isProfilingJobPaused) {
+      resumeProfilingJob()
+    }
+  }
```

Comment:
This try/catch would never throw as the only condition in which `pprof.time.stop()` can throw is already covered by the `ProfilingNotStartedError` use above.